tests/clv/models/test_beta_geo.py

```diff
@@ -324,3 +398,32 @@ def test_model_repr(self):
             "\nr~HalfFlat()"
             "\nlikelihood~Potential(f(r,alpha,b,a))"
         )
+
+    def test_save_load_beta_geo(self, data):
+        temp = tempfile.NamedTemporaryFile(mode="w", encoding="utf-8", delete=False)
+
+        model = BetaGeoModel(
+            data=data,
+        )
+        model.build_model()
+        model.fit("map")
+        model.save(temp)
+        # Testing the valid case.
+
+        model2 = BetaGeoModel.load(temp)
+
+        # Check if the loaded model is indeed an instance of the class
+        assert isinstance(model, BetaGeoModel)
+
+        # Load data from the file to cross verify
+        filepath = Path(str(temp))
+        idata = az.from_netcdf(filepath)
+        dataset = idata.fit_data.to_dataframe()
+        # Check if the loaded data matches with the model data
+        assert np.array_equal(model2.customer_id.values, dataset.customer_id.values)
+        assert np.array_equal(model2.frequency.values, dataset.frequency.values)
+        assert np.array_equal(model2.T.values, dataset["T"])
```

Comment:
Should use np.testing utilities, as they provide much better message when it fails

Reply:
replaced